source/windows/secure_channel_tls_handler.c

```diff
@@ -1252,7 +1259,10 @@ static int s_process_read_message(
     struct aws_io_message *message) {
 
     struct secure_channel_handler *sc_handler = handler->impl;
-    if (sc_handler->read_shutdown_completed) {
+    if (sc_handler->read_state == AWS_TLS_HANDLER_READ_SHUT_DOWN_COMPLETE) {
+        if (message) {
+            aws_mem_release(message->allocator, message);
+        }
         return AWS_OP_SUCCESS;
     }
 

```

Comment:
down below, on line 1305, which is like:
```
/* throw this one as a protocol error. */
aws_raise_error(AWS_IO_TLS_ERROR_WRITE_FAILURE);
```

I think there should be a `break;` afterwards?